File: 538-convert-bst-to-greater-tree/538-convert-bst-to-greater-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def convertBST(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        
        def postorder(root):
            if root is None:
                return 0
            
            postorder(root.right)
            postorder.static_variable += root.val
            root.val = postorder.static_variable
            postorder(root.left)
        
        postorder.static_variable = 0
        postorder(root)
        return root

File: 538-convert-bst-to-greater-tree/538-convert-bst-to-greater-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */    
struct TreeNode* convertBST(struct TreeNode* root){
    int sum_val = 0;
    int newOrder(struct TreeNode* root){
        
        if(root==NULL)
        return 0;
        
        newOrder(root->right);
        
        sum_val = sum_val + root->val;
        root->val = sum_val;
        
        newOrder(root->left);
        
        return 0;
    }
    
    newOrder(root);
    return root;
    
}